Realm/RLMSyncUser.h

```diff
@@ -113,8 +140,118 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
 - (NSArray<RLMSyncSession *> *)allSessions;
 
 /**
+ Given the URL of a Realm managed by this user, modify permissions of
+ that Realm for a different user.
+
+ If no URL is passed in, the permission changes will be applied to ALL
+ synced Realms managed by this user.
+
+ If no user is passed in, the permission changes will be applied with
+ respect to the given synced Realm (or all synced Realms) for ALL users.
+
+ @param realmURL        The URL of the synced Realm on the server whose
+                        permissions should be modified, or nil.
+ @param identity        The identity of the user whose permissions for
+                        the Realm at `realmURL` should be modified, or
+                        `nil`.
+ @param permissions     The new permissions to be set.
+ @param callback        An optional block through which the progress of
+                        the permission change operation can be reported.
+ @returns   If `callback` was specified, a notification token. Hold onto
+            the token for as long as callback status updates are desired.
+            Call `-stop` on the token before destroying it.
+ */
+- (nullable RLMNotificationToken *)modifyRealmPermissionsForURL:(nullable NSURL *)realmURL
```

Comment:
I don't think we should require a notification token. This is a case where the callback should get called without the need for a token.